### PR TITLE
Move small key messages to new system and include info about what's unlocked by silver rupees

### PR DESCRIPTION
**Pull request by fenhl**
This also adjusts the messages so they no longer show the count if you're sending one of these items to another world.

### CODE REVIEW

File: ASM/c/item_table.c [hunk 1]
```diff
@@ -309,22 +309,67 @@ item_row_t item_table[] = {
...
+        // The player already has enough silver rupees.
+        return item_row->text_id + 0x16;
+    } else if (!is_outgoing && extended_savectx.silver_rupee_counts[silver_rupee_id] + 1 == var.needed_count) {
+        // This silver rupee unlocks the puzzle.
+        return item_row->text_id + (CFG_DUNGEON_IS_MQ[dungeon_id] ? 0x2E : 0x44);
+    } else if (!is_outgoing && extended_savectx.silver_rupee_counts[silver_rupee_id] > 0) {
+        // Show number collected.
+        return item_row->text_id;
+    } else {
+        // This silver rupee is for another player or the first of this puzzle, don't show any extra info.
```
Comment (rrealmuto) [inline ASM/c/item_table.c:332]: Don't think I agree with this change. I think it's more consistent to just always show the count if it's for this player.

**Review by rrealmuto — COMMENTED** [review]
"*" goes to the left with the type

File: ASM/c/item_table.c (hunk 1 above)
Comment (fenhl) [reply ASM/c/item_table.c:332]: This is how it works for small keys as well.

Comment (rrealmuto) [reply ASM/c/item_table.c:332]: Yeah I noticed. Thought it was broken until I collected a second one...

**Pull request merged**